packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx

```diff
@@ -224,12 +219,66 @@ const TradeParamsMobile = observer(
         h_duration,
         d_duration,
     }: TTradeParamsMobile) => {
-        const { basis_list, basis, expiry_epoch, is_turbos, is_vanilla } = useTraderStore();
+        const {
+            basis_list,
+            basis,
+            expiry_epoch,
+            is_turbos,
+            is_vanilla,
+            duration_unit: store_duration_unit,
+        } = useTraderStore();
+
+        const resetToDefaultDuration = () => {
+            const default_store_duration = getDefaultDuration(store_duration_unit);
+            setSelectedDuration(store_duration_unit, default_store_duration);
+        };
+
+        React.useEffect(() => {
+            const toggled_duration_tab_idx = duration_units_list.findIndex(d => d.value === duration_unit);
+            const default_duration_tab_idx = duration_units_list.findIndex(d => d.value === store_duration_unit);
+            const default_duration = getDefaultDuration(duration_unit);
+
+            // reset to default value and timeframe when previously chosen timeframe doesnt exist in new trade type
```

Comment:
```suggestion
            // reset to default value and time frame when previously chosen time frame doesn't exist in new trade type
```
nit-pick)

Reply:
Timeframe is the correct version though 😄